cli/npm/cache_dir.rs

```diff
@@ -97,23 +104,37 @@ impl NpmCacheDir {
   pub fn resolve_package_folder_id_from_specifier(
     &self,
     specifier: &ModuleSpecifier,
-    registry_url: &Url,
   ) -> Option<NpmPackageCacheFolderId> {
-    let registry_root_dir = self
-      .root_dir_url
-      .join(&format!(
-        "{}/",
-        root_url_to_safe_local_dirname(registry_url)
-          .to_string_lossy()
-          .replace('\\', "/")
-      ))
-      // this not succeeding indicates a fatal issue, so unwrap
-      .unwrap();
-    let mut relative_url = registry_root_dir.make_relative(specifier)?;
-    if relative_url.starts_with("../") {
-      return None;
```

Comment:
I think we could make this more optimal by storing `root_url_to_safe_local_dirname` for each registry url in the constructor, then here getting the relative path from the `self.root_dir_url`, then check the first component for a match, then continue below